### ISSUE TITLE
post test results to publicly available URL

### ISSUE THREAD

**Issue by gregcaporaso** [caporaso-lab-graveyard/clout#2]
Would be really cool if the tests results that get emailed also were posted to the internet somewhere. Could use GitHub pages for this. If we were going to do that, we could probably also post updates to the developer documentation on a nightly basis.

**Comment by wasade** [comment]
Agree

**Comment by jairideout** [comment]
Looking into this a bit- what do you guys think about posting testing results on the [Clout wiki](https://github.com/qiime/clout/wiki)? We could have a section for each project, and each section would have a list ordered by date with links to the results.

If other people started using Clout, they could optionally post their testing results to their own wikis.

**Comment by jairideout** [comment]
Some more info: the current size of the nightly test results is around 0.5MB. If we were to use the wiki to host these results, it would take us around 5.5 years to reach GitHub's 1GB recommended repo size limit. Seems pretty reasonable, and if we start running out of space, we could always archive older test results.

**Comment by wasade** [comment]
0.5MB tar gz'd?

**Comment by wasade** [comment]
ah...

**Comment by jairideout** [comment]
0.5MB not zipped (7 plain text files, excluding the complete log). We could store them in .tar.gz format, but I was thinking it'd be cool if they could be directly viewable online from within a web browser.

**Comment by gregcaporaso** [comment]
I definitely think it'd be good to have them viewable from a browser. What about storing them on the AWS-based web server that you're going to build?

**Comment by jairideout** [comment]
We could do that, but I like the wiki idea better, since Clout and most of the projects that it's testing are on GitHub already. That way, the testing results will also be on GitHub, under version control (in the wiki repo, which is separate from the main one), and storage/hosting is free.

We could add the ability to automatically post results to a GitHub wiki, or (optionally) store them on the machine that runs Clout.

Do you see issues with the wiki approach versus hosting them from our new web server?

**Comment by gregcaporaso** [comment]
Would the user specify which project they live under? I don't think we'd want other people's projects to show up under clout. I do worry a little about making it too GitHub specific.

**Comment by jairideout** [comment]
In the config file there'd be a way to (optionally) specify what wiki (and where) to place the results. We'd most likely add results for all of the current projects being tested on a nightly basis (e.g. QIIME, PyCogent, etc.) since they are all related bioinformatics software.

If there's an option to support a GitHub wiki and/or just save to disk, would this get around the issue of being too GitHub-specific?

**Comment by gregcaporaso** [comment]
> If there's an option to support a GitHub wiki and/or just save to disk, would this get around the issue of being too GitHub-specific?

Yes, definitely - then they could just define another cron job to poll for new files and do something with them.

**Comment by gregcaporaso** [comment]
So I think you should go ahead with this solution.

**Comment by jairideout** [comment]
Great, thanks for the feedback!
